lib/src/dart3_suggestors/required_props/bin/collect.dart

```diff
@@ -0,0 +1,315 @@
+// Copyright 2024 Workiva Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import 'dart:async';
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:args/command_runner.dart';
+import 'package:logging/logging.dart';
+import 'package:over_react_codemod/src/dart3_suggestors/required_props/collect/aggregate.dart';
+import 'package:over_react_codemod/src/dart3_suggestors/required_props/collect/analysis.dart';
+import 'package:over_react_codemod/src/dart3_suggestors/required_props/collect/collect.dart';
+import 'package:over_react_codemod/src/util/command.dart';
+import 'package:over_react_codemod/src/util/command_runner.dart';
+import 'package:package_config/package_config.dart';
+import 'package:path/path.dart' as p;
+import 'package:yaml/yaml.dart';
+
+import '../collect/collected_data.sg.dart';
+import '../collect/logging.dart';
+import '../collect/package/parse_spec.dart';
+import '../collect/package/spec.dart';
+import '../collect/package/version_manager.dart';
+
+class CollectCommand extends Command {
+  @override
+  String get description =>
+      'Collects requiredness data for all OverReact props based on usages in the specified packages and all their transitive dependencies.';
+
+  @override
+  String get name => 'collect';
+
+  @override
+  String get invocation =>
+      '$invocationPrefix [<options>] <package_spec> [<package_spec>...]';
+
+  @override
+  String get usageFooter =>
+      '\n$packageSpecFormatsHelpText\n\n$_usageInstructions';
+
+  String get _usageInstructions => '''
+Instructions
+============
+
+1. First, identify the least-common consumer(s) of OverReact components exposed by your package.
+
+   (If all your package's components are private, you can skip the rest of this step,
+   step and just use your package).
+
+   For example, say we're dealing with package A, which is directly consumed by
+   packages B, E, and F, and so on:
+   
+       ${r'A---B---C---D'}
+       ${r'|\     /'}
+       ${r'| E----'}
+       ${r'\'}
+       ${r' F---G---H'}
+
+   The least-common consumers would be C (covers both B and E) and F, so we'd run:
+   
+      $invocationPrefix pub@…:C pub@…:F
+   
+   Note: if F were to re-export members of A, which could potentially get used
+   in G, we'd do G instead of F.
```

Comment:
Sorry I still don't get this. I understand the last `Alternatively,...` would just be doing `wdesk` or something, but I don't really understand what on the graph makes it okay to do C and F

Reply:
Sorry about that; I was trying to capture that there may be branches of downstream consumers that aren't related. For instance, in this example, there's no package that transitively consumes both C and F.

Was that the part you weren't following, or maybe something else? Happy to hop on a call and talk through it too

And I'm totally down to adjust this example however needed to make things clearer

Reply:
Yeah that makes sense, I guess I just don't get why it can be F and C and not D and H

Reply:
Oh it can be D and H, the idea is just that F and C is preferred (mainly because we get to skip analyzing code for D, H, G, and any other transitive dependencies not pulled in by F or C)

Reply:
...And also increases the risk of mismatched dependencies talked about in the next step:

> 2. If step 1 yielded more than one package, make sure all of them can resolve to
>   the latest version of your package. 
>   
>   If they can't, then data may be missing for recently-added props, or could be
>   incorrect if props in your package were moved to different files.

Reply:
oh wait I get it - I was misunderstanding the transitive dependency thing and that D and H aren't directly consuming A - cool cool that makes sense! Thank you!

Reply:
Sweet, np! If you have any suggestions on how to make the wording or diagram more clear, I'm all ears!